File: safelima/lib/models/ml_model.dart
import 'dataset.dart';

class MlModel {
  final int? id;
  final String? nombreModelo;
  final int? datasetId;
  final String? version;
  final String? rutaModelo;
  final double? precision;
  final double? accuracy;
  final double? recall;
  final double? f1;
  final double? auc;
  final DateTime? fechaEntrenamiento;
  final Dataset? dataset;

  MlModel({
    this.id,
    this.nombreModelo,
    this.datasetId,
    this.version,
    this.rutaModelo,
    this.precision,
    this.accuracy,
    this.recall,
    this.f1,
    this.auc,
    this.fechaEntrenamiento,
    this.dataset,
  });

  factory MlModel.fromJson(Map<String, dynamic> json) {
    return MlModel(
      id: json['id'],
      nombreModelo: json['nombre_modelo'],
      datasetId: json['dataset_id'],
      version: json['version'],
      rutaModelo: json['ruta_modelo'],
      precision: (json['precision'] as num?)?.toDouble(),
      accuracy: (json['accuracy'] as num?)?.toDouble(),
      recall: (json['recall'] as num?)?.toDouble(),
      f1: (json['f1'] as num?)?.toDouble(),
      auc: (json['auc'] as num?)?.toDouble(),
      fechaEntrenamiento: json['fecha_entrenamiento'] != null
          ? DateTime.parse(json['fecha_entrenamiento'])
          : null,
      dataset: json['dataset'] != null
          ? Dataset.fromJson(json['dataset'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nombre_modelo': nombreModelo,
      'dataset_id': datasetId,
      'version': version,
      'ruta_modelo': rutaModelo,
      'precision': precision,
      'accuracy': accuracy,
      'recall': recall,
      'f1': f1,
      'auc': auc,
      'fecha_entrenamiento': fechaEntrenamiento?.toIso8601String(),
    };
  }
}

File: safelima_app/app/models/ml_model.py
from sqlalchemy import Column, Integer, String, Text, Numeric, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database.database import Base

class MLModel(Base):
    __tablename__ = "ml_models"

    id = Column(Integer, primary_key=True, index=True)
    nombre_modelo = Column(String(100))
    dataset_id = Column(Integer, ForeignKey("datasets.id", ondelete="SET NULL"))
    version = Column(String(20))
    ruta_modelo = Column(Text)
    precision = Column(Numeric(5, 2))
    accuracy = Column(Numeric(5, 2))
    recall = Column(Numeric(5, 2))
    f1 = Column(Numeric(5, 2))
    auc = Column(Numeric(5, 2))
    fecha_entrenamiento = Column(DateTime, default=datetime.utcnow)

    # Relaciones
    dataset = relationship("Dataset", back_populates="models")
